core/src/repair_service.rs

```diff
@@ -200,30 +200,31 @@ impl RepairService {
     ) -> Result<(Vec<RepairType>)> {
         // Slot height and blob indexes for blobs we want to repair
         let mut repairs: Vec<RepairType> = vec![];
-        let mut meta_iter = blocktree
-            .slot_meta_iterator(repair_range.start)
-            .expect("Couldn't get db iterator");
-        while repairs.len() < max_repairs && meta_iter.valid() {
-            let current_slot = meta_iter.key();
-            if current_slot.unwrap() > repair_range.end {
+        for slot in repair_range.start..=repair_range.end {
+            if repairs.len() >= max_repairs {
                 break;
             }
 
-            if current_slot.unwrap() > repair_info.max_slot {
+            if slot > repair_info.max_slot {
```

Comment:
I think we can get rid of this `max_slot` logic and the optimistic trying to repair max_slot + 1 as the end of this function, since we know exactly what the range of slots we are trying to repair is